src/lib/coerce.js

```diff
@@ -257,19 +257,56 @@ exports.valObjectMeta = {
             'An {array} of plot information.'
         ].join(' '),
         requiredOpts: ['items'],
-        otherOpts: ['dflt', 'freeLength'],
+        // set dimensions=2 for a 2D array
+        // `items` may be a single object instead of an array, in which case
+        // `freeLength` must be true.
```

Comment:
I needed 1D and 2D arbitrary-length `info_array` for specifying axes and/or subplots for the grid. I didn't actually end up using `coerce` with these, since there's validation involved where allowed values depend on both the available axes/subplots and on earlier elements in the same array; but it's still important for documentation, and for `Plotly.react` it's important that these are marked as `info_array` and not `data_array`.

Anyway, they work, in case we find a need in the future... tested below.

Reply:
These ones are for function arguments, and we know that the functions being called have a maximum of 3 arguments. The `[{}, {}, {}]` form ensures we never go past three args. I suppose it *probably* wouldn't hurt to change it, but I didn't feel like taking that risk as I'm not sure how well tested the edge cases are for these components.

Reply:
Good point here about `args` being at-most a three-item array. Let's keep that as it :ok_hand: 

But this has me thinking: should we change attributes like `ticktext` and `tickvals` to free-length `info_array` instead of their current `data_array` val type? That way, we could get rid of this ugly [hack](https://github.com/plotly/plotly.js/blob/1256c4f9de37ad68ba82d28a048dbe96bc204c4d/src/plot_api/plot_schema.js#L180).

Reply:
> should we change attributes like `ticktext` and `tickvals` to free-length `info_array` instead of their current `data_array` val type?

Hmm, possibly... would certainly be nice to keep `data_array` out of `layout`, but those are potentially long enough (and likely enough to be connected to a data source) to still warrant being called `data_array` so I'm a bit ambivalent about it. We could discuss this in a separate issue.

Reply:
> We could discuss this in a separate issue.

Sounds good :+1:

Reply:
Let's continue that discussion in -> https://github.com/plotly/plotly.js/issues/1894